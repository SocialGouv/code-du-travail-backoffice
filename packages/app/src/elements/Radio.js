/* eslint-disable max-len */

import styled from "@emotion/styled";
import PropTypes from "prop-types";
import React from "react";

import Icon from "./Icon";

const Bullet = styled(Icon)`
  color: var(--color-text-blue);
  cursor: ${({ selected }) => (!selected ? "pointer" : "auto")};
  margin-right: 0.5rem;

  :hover {
    color: ${({ selected }) =>
      !selected ? "var(--color-lapis-lazuli)" : "var(--color-text-blue)"};
  }
`;

const Radio = ({ onChange, options, ...props }) => (
  <div role="radiogroup">
    {options.map(({ isSelected = false, label, value }, index) => (
      <div key={index}>
        <Bullet
          aria-checked={String(isSelected)}
          aria-hidden="false"
          icon={`${isSelected ? "dot-" : ""}circle`}
          onClick={!isSelected ? () => onChange(value) : undefined}
          role="radio"
          selected={isSelected}
          tabIndex={index === 0 ? "0" : "-1"}
          {...props}
        />
        {label}
      </div>
    ))}
  </div>
);

Radio.propTypes = {
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.exact({
      isSelected: PropTypes.bool,
      label: PropTypes.string,
      value: PropTypes.string,
    }),
  ),
};

export default Radio;
