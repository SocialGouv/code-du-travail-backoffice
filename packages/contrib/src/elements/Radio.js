/* eslint-disable max-len */

import React from "react";
import styled from "@emotion/styled";

import _Icon from "./Icon";

const Icon = styled(_Icon)`
  color: var(--color-text-blue);
  cursor: ${({ selected }) => (!selected ? "pointer" : "auto")};
  margin-right: 0.5rem;

  :hover {
    color: ${({ selected }) =>
      !selected ? "var(--color-lapis-lazuli)" : "var(--color-text-blue)"};
  }
`;

export default ({ onChange, options, ...props }) => {
  return options.map(({ isSelected, label, value }, index) => (
    <div key={index}>
      <Icon
        icon={`${isSelected ? "dot-" : ""}circle`}
        onClick={() => onChange(value)}
        selected={isSelected}
        {...props}
      />
      {label}
    </div>
  ));
};
