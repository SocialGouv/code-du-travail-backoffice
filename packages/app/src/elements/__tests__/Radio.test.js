import React from "react";

import Radio from "../Radio";

describe("elements/<Radio />", () => {
  const COMMON_PROPS = {
    options: [
      { isSelected: true, label: "A Selected Label", value: "A Selected Value" },
      { label: "A Label", value: "A Value" },
      { label: "Another Label", value: "Another Value" }
    ]
  };

  it(`should pass`, () => {
    const $radio = testRender(<Radio {...COMMON_PROPS} />);
    const $firstSvg = $radio.children[0].findByType("svg");
    const $secondSvg = $radio.children[1].findByType("svg");

    expect($radio).toMatchSnapshot();
    expect($radio.props).toHaveProperty("role", "radiogroup");
    expect($firstSvg.props).toHaveProperty("aria-checked", "true");
    expect($secondSvg.props).toHaveProperty("aria-hidden", "false");
    expect($firstSvg.props).toHaveProperty("data-icon", "dot-circle");
    expect($firstSvg.props).toHaveProperty("tabIndex", "0");
    expect($firstSvg.props).toHaveProperty("role", "radio");
    expect($secondSvg.props).toHaveProperty("aria-checked", "false");
    expect($secondSvg.props).toHaveProperty("aria-hidden", "false");
    expect($secondSvg.props).toHaveProperty("data-icon", "circle");
    expect($secondSvg.props).toHaveProperty("tabIndex", "-1");
    expect($secondSvg.props).toHaveProperty("role", "radio");
  });

  it(`should not call {onChange} when the selected radio is clicked`, () => {
    const props = {
      onChange: jest.fn()
    };
    const $radio = testRender(<Radio {...props} {...COMMON_PROPS} />);
    const $firstSvg = $radio.children[0].findByType("svg");

    $firstSvg.props.onClick();
    expect(props.onChange).not.toHaveBeenCalled();
  });

  it(`should call {onChange} when a unselected radio is clicked`, () => {
    const props = {
      onChange: jest.fn()
    };
    const $radio = testRender(<Radio {...props} {...COMMON_PROPS} />);
    const $secondSvg = $radio.children[1].findByType("svg");

    $secondSvg.props.onClick();
    expect(props.onChange).toHaveBeenCalledWith("A Value");
  });
});
