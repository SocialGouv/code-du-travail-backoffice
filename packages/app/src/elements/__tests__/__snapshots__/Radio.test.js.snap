// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/<Radio /> should pass 1`] = `
.emotion-1 {
  cursor: default;
  width: 1rem !important;
  color: var(--color-text-blue);
  cursor: auto;
  margin-right: 0.5rem;
}

.emotion-1:hover {
  color: var(--color-text-blue);
}

.emotion-4 {
  cursor: default;
  width: 1rem !important;
  color: var(--color-text-blue);
  cursor: pointer;
  margin-right: 0.5rem;
}

.emotion-4:hover {
  color: var(--color-lapis-lazuli);
}

<div
  role="radiogroup"
>
  <div>
    <svg
      aria-checked="true"
      aria-hidden="false"
      className="svg-inline--fa fa-dot-circle fa-w-16 fa-1x emotion-0 emotion-1 emotion-2"
      color="var(--color-black-leather-jacket)"
      data-icon="dot-circle"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="radio"
      selected={true}
      style={Object {}}
      tabIndex="0"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    A Selected Label
  </div>
  <div>
    <svg
      aria-checked="false"
      aria-hidden="false"
      className="svg-inline--fa fa-circle fa-w-16 fa-1x emotion-0 emotion-4 emotion-2"
      color="var(--color-black-leather-jacket)"
      data-icon="circle"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="radio"
      selected={false}
      style={Object {}}
      tabIndex="-1"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    A Label
  </div>
  <div>
    <svg
      aria-checked="false"
      aria-hidden="false"
      className="svg-inline--fa fa-circle fa-w-16 fa-1x emotion-0 emotion-4 emotion-2"
      color="var(--color-black-leather-jacket)"
      data-icon="circle"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="radio"
      selected={false}
      style={Object {}}
      tabIndex="-1"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Another Label
  </div>
</div>
`;
